Add CocoaPods downloads badges.

nlutsenko opened the pull request:
Expands CocoaPods integration using `metrics.cocoapods.org`.
Tried my best to match the existing style for other package managers.

Also cc @orta @segiddins @kylef as core CocoaPods people.

nlutsenko commented:
Also, tested locally and this is the output :grin:
![screen shot 2016-05-04 at 7 37 03 pm](https://cloud.githubusercontent.com/assets/606991/15034881/9a8e409a-122f-11e6-9bfa-e6b94bc37d05.png)

segiddins commented:
@nlutsenko perhaps these should use apps rather than download counts?

nlutsenko commented:
I think we should do both, tbh, but not sure where to put 2 more metrics for `apps` and what would the badge look like?

nlutsenko commented:
@espadrine, does this look reasonable to you? Would love to have this merged!

orta commented:
I'd also love to see the apps in there, FWIW, far more interesting metadata 💃 (though usually significantly smaller number :D )

nlutsenko commented:
Very well, then. Let me add `apps` here as well. :D
Update to the PR is incoming.

nlutsenko commented:
Perfect, this is updated now and has `apps` badges, per popular request.
And, of course, here is a fancy and visually appealing screenshot 
![screen shot 2016-05-11 at 9 55 48 pm](https://cloud.githubusercontent.com/assets/606991/15204583/9bbc3ac8-17c3-11e6-85e4-8af166658339.png)

This is also rebased on the latest master, cleaned, rinsed and ready for review.

orta commented:
👍 thanks @nlutsenko

nlutsenko commented:
@espadrine, if I rebase it - would you want to merge this one in?
We have a buy-in from the CocoaPods core team, and it's a pretty useful addition to the shields list, and can increase everyone's ego growth by some percentage (or at least I very much hope so).

espadrine commented on server.js:
```diff
@@ -2719,6 +2719,84 @@ cache(function(data, match, sendBadge, request) {
+  var spec = match[2];  // eg, AFNetworking
+  var format = match[3];
+  var apiUrl = 'http://metrics.cocoapods.org/api/v1/pods/' + spec;
+  var badgeData = getBadgeData('pod', data);
```
'downloads' instead of 'pod'. It incidentally removes the need for `getLabel()` below.

espadrine commented on server.js:
```diff
@@ -2719,6 +2719,84 @@ cache(function(data, match, sendBadge, request) {
+  var spec = match[2];  // eg, AFNetworking
+  var format = match[3];
+  var apiUrl = 'http://metrics.cocoapods.org/api/v1/pods/' + spec;
+  var badgeData = getBadgeData('pod', data);
```
'apps' instead of 'pod', which also removes the need for `getLabel()` below.

espadrine commented on server.js:
```diff
@@ -2719,6 +2719,84 @@ cache(function(data, match, sendBadge, request) {
+    try {
+      var data = JSON.parse(buffer);
+      badgeData.text[0] = getLabel('apps', data);
+      var apps = 0
```
Missing semicolon.

espadrine commented on server.js:
```diff
@@ -2719,6 +2719,84 @@ cache(function(data, match, sendBadge, request) {
+      var apps = 0
+      switch (info.charAt(1)) {
+        case 'w':
+          var apps = data.stats.app_week;
```
No need to define `apps` with var, as it was already defined in the function.

espadrine commented:
Thanks a lot for the contribution! I applied the adjustments from my review myself, as they are rather small.

It is live, but I'll update the website in 24h. You can check that everything works fine in the meantime.
